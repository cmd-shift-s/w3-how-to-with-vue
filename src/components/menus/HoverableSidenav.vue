<template lang="html">
  <div class="hoverable-sidenav">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'hoverable-sidenav'
}
</script>

<style lang="scss">
/* Style the links inside the sidenav */
.hoverable-sidenav {
  display: flex;
  flex-direction: column;
  // overflow: hidden;

  a {
    position: relative; /* Position them relative to the browser window */
    left: 0; /* Position them outside of the screen */
    transition: 0.3s; /* Add transition on hover */
    padding: 15px; /* 15px padding */
    margin: 5px 0;
    margin-left: -90px;
    width: 100px; /* Set a specific width */
    text-decoration: none; /* Remove underline */
    font-size: 20px; /* Increase font size */
    color: white; /* White text color */
    border-radius: 0 5px 5px 0; /* Rounded corners on the top right and bottom right side */

    &:hover {
      left: 80px;
    }
  }
}
</style>
