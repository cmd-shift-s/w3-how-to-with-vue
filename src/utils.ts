import MarkdownIt from 'markdown-it'
import hljs from 'highlight.js'
import 'highlight.js/styles/vs.css'

export const md = new MarkdownIt({
  highlight (code, lang) {
    return hljs.highlight(lang, code).value
  }
})
