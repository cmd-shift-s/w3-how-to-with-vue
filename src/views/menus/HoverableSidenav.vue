<template lang="html">
  <page
    title="Side Navigation Buttons"
    intro="hoverable side navigation buttons with CSS">
    <sample class="hoverable-sidenav-page">
      <hoverable-sidenav>
        <a style="background-color: green;">About</a>
        <a style="background-color: blue;">Blog</a>
        <a style="background-color: red;">Projects</a>
        <a style="background-color: black;">Contact</a>
      </hoverable-sidenav>
      <section class="contents">
        <h2>Hoverable Sidenav Buttons</h2>
        <p>Hover over the buttons in the left side navigation to open them.</p>
      </section>
    </sample>

    <template slot="examples">
    <hr>
    <h2 class="title">How To Create a Hoverable Sidenav</h2>
    <step no="1" html></step>
    <example-code :code="examples[0]"></example-code>
    <step no="2" css></step>
    <example-code :code="examples[1]"></example-code>
    </template>
  </page>
</template>

<script>
import HoverableSidenav from '@/components/menus/HoverableSidenav.vue'
export default {
  name: 'hoverable-sidenav-page',
  data() {
    return {
      examples: [
        [
          '``` html',
          '<div id="mySidenav" class="sidenav">',
          '  <a href="#" id="about">About</a>',
          '  <a href="#" id="blog">Blog</a>',
          '  <a href="#" id="projects">Projects</a>',
          '  <a href="#" id="contact">Contact</a>',
          '</div>',
          '```'
        ],
        [
          '``` css',
          '/* Style the links inside the sidenav */',
          '#mySidenav a {',
          '  position: absolute; /* Position them relative to the browser window */',
          '  left: -80px; /* Position them outside of the screen */',
          '  transition: 0.3s; /* Add transition on hover */',
          '  padding: 15px; /* 15px padding */',
          '  width: 100px; /* Set a specific width */',
          '  text-decoration: none; /* Remove underline */',
          '  font-size: 20px; /* Increase font size */',
          '  color: white; /* White text color */',
          '  border-radius: 0 5px 5px 0; /* Rounded corners on the top right and bottom right side */',
          '}',
          '',
          '#mySidenav a:hover {',
          '  left: 0; /* On mouse-over, make the elements appear as they should */',
          '}',
          '',
          '/* The about link: 20px from the top with a green background */',
          '#about {',
          '  top: 20px;',
          '  background-color: #4CAF50;',
          '}',
          '',
          '#blog {',
          '  top: 80px;',
          '  background-color: #2196F3; /* Blue */',
          '}',
          '',
          '#projects {',
          '  top: 140px;',
          '  background-color: #f44336; /* Red */',
          '}',
          '',
          '#contact {',
          '  top: 200px;',
          '  background-color: #555 /* Light Black */',
          '}',
          '```'
        ]
      ]
    }
  },
  components: {
    HoverableSidenav
  }
}
</script>

<style lang="scss">
.hoverable-sidenav-page {
  border: 1px solid #ccc;
  padding: 20px 0;
  overflow: hidden;

  a {
    cursor: pointer;
  }

  .contents {
    margin-left: 30px;
  }
}
</style>
