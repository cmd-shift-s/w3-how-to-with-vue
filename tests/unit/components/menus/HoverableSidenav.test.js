import { mount, renderer } from '@/tests/unit/utils'
import HoverableSidenav from '@/components/menus/HoverableSidenav.vue'

describe('HoverableSidenav.vue', () => {
  it('render default', () => {
    const wrap = mount(HoverableSidenav)

    expect(wrap.classes()).toContain('hoverable-sidenav')

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
