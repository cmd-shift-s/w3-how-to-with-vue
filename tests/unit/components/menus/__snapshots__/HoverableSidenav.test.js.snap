// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverableSidenav.vue render default 1`] = `<div class="hoverable-sidenav"></div>`;
